frontend/src/Sidebar/Sidebar.test.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import Sidebar from './Sidebar';
+
+const topics = [{ _id: '1', name: 'Voyage', order: 1 }];
+// eslint-disable-next-line camelcase
+const sub_topics = [
+  { _id: '10', parent: '1', name: 'About this Wiki', order: 1 },
+  { _id: '11', parent: '1', name: 'About Voyages', order: 2 },
+];
+
+const articles = [
+  {
+    _id: '100',
+    topic: topics[0],
+    sub_topic: sub_topics[0],
+    title: 'Home',
+    order: 1,
+  },
+  {
+    _id: '101',
+    topic: topics[0],
+    sub_topic: sub_topics[0],
+    title: 'How to Contribute',
+    order: 2,
+  },
+  {
+    _id: '102',
+    topic: topics[0],
+    sub_topic: sub_topics[1],
+    title: 'Voyage Roadmap',
+    order: 1,
+  },
+  {
+    _id: '103',
+    topic: topics[0],
+    sub_topic: null,
+    title: 'Introduction',
+    order: 1,
+  },
+];
+
+const expectedTree = {
+  Voyage: {
+    _id: '1',
+    expanded: false,
+    'About this Wiki': {
+      _id: '10',
+      expanded: false,
+      Home: '100',
+      'How to Contribute': '101',
+    },
+    'About Voyages': {
+      _id: '11',
+      expanded: false,
+      'Voyage Roadmap': '102',
+    },
+    Introduction: '103',
+  },
+};
+
+const expectedExpandedTree = {
+  Voyage: {
+    _id: '1',
+    expanded: true,
+    'About this Wiki': {
+      _id: '10',
+      expanded: false,
+      Home: '100',
+      'How to Contribute': '101',
+    },
+    'About Voyages': {
+      _id: '11',
+      expanded: false,
+      'Voyage Roadmap': '102',
+    },
+    Introduction: '103',
+  },
+};
+
+const match = {
+  params: { id: '' },
+};
+
+function flushPromises() {
+  return new Promise(resolve => setImmediate(resolve));
+}
+
+let comp;
+let inst;
+
+it('should initialise Sidebar correctly', () => {
+  comp = shallow(<Sidebar articles={articles} match={match} />);
+  inst = comp.instance();
+  comp.setProps({ articles });
+  expect(inst.state.open).toBe(false);
+  expect(inst.state.articleTree).toEqual(expectedTree);
+  expect(inst.state.articlesHtml).toMatchSnapshot();
+});
+
+it('should show mobile drawer', () => {
+  expect.hasAssertions();
+  comp.setState({ articlesHtml: null, mobile: true });
+  inst.openDrawer();
+  return Promise.resolve().then(() => {
+    comp.update();
+    expect(inst.state.open).toBe(true);
+    expect(comp).toMatchSnapshot();
+  });
+});
```

Comment:
Passing of this test depends on the previous test. Not good.

Reply:
It seemed natural for one test to run on the results of the last - I take it this is bad practice. Will fix

Reply:
I don't think it's natural.
Better way to accomplish that would be:
```js
it('should test some stuff', () => {
  // initial setup
  // assert that setup is correct
  // simulate user interaction, e.g. click
  // assert changes
  // simulate another interaction
  // assert changes
  // etc.
}
```

Reply:
@jenovs is this comment out of order? you're saying tests following on isn't natural ... or do you mean the beforeEach isn't the best way to fix?

Reply:
oh I see you mean put them in the same test - ok. Will merge shortly

Reply:
Looks like we wrote our comments almost simultaneously :grinning: 
I edited my comment.